tests/integration/test_s3.py

```diff
@@ -864,6 +864,14 @@ def test_delete_non_existing_keys(self):
         # clean up
         self._delete_bucket(bucket_name)
 
+    def test_delete_non_existing_keys_in_non_existing_bucket(self):
+        with self.assertRaises(ClientError) as ctx:
+            self.s3_client.delete_objects(
```

Comment:
We could consider testing this for the two main S3 addressing types - path-based addressing, and hostname-based addressing (see [here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html)).

Actually - there may be an opportunity for an overall test refactoring to achieve that - will cover it in a separate work item.. (no need to change for this PR then) 👍

Reply:
For the reference https://github.com/localstack/localstack/issues/4952